lib/pbench/test/unit/server/test_visualize.py

```diff
@@ -0,0 +1,116 @@
+from http import HTTPStatus
+from pathlib import Path
+
+from pquisby.lib.post_processing import QuisbyProcessing
+import pytest
+import requests
+
+from pbench.server.api.resources import ApiBase
+from pbench.server.cache_manager import CacheManager
+from pbench.server.database.models.datasets import Dataset, DatasetNotFound
+
+
+class TestVisualize:
+    @pytest.fixture()
+    def query_get_as(self, client, server_config, more_datasets, get_token_func):
+        """
+        Helper fixture to perform the API query and validate an expected
+        return status.
+
+        Args:
+            client: Flask test API client fixture
+            server_config: Pbench config fixture
+            more_datasets: Dataset construction fixture
+            get_token_func: Pbench token fixture
+        """
+
+        def query_api(
+            dataset: str, user, expected_status: HTTPStatus
+        ) -> requests.Response:
+            try:
+                dataset_id = Dataset.query(name=dataset).resource_id
+            except DatasetNotFound:
+                dataset_id = dataset  # Allow passing deliberately bad value
+            headers = {"authorization": f"bearer {get_token_func(user)}"}
+            response = client.get(
+                f"{server_config.rest_uri}/visualize/{dataset_id}",
+                headers=headers,
+            )
+            assert response.status_code == expected_status
+            return response
+
+        return query_api
+
+    def mock_find_dataset(self, dataset):
+        class Tarball(object):
+            tarball_path = Path("/dataset/tarball.tar.xz")
+
+            def extract(tarball_path, path):
+                return "CSV_file_as_a_byte_stream"
+
+        return Tarball
+
+    def mock_get_dataset_metadata(self, dataset, key):
+        return {"dataset.metalog.pbench.script": "uperf"}
+
+    def test_get_no_dataset(self, query_get_as):
+        response = query_get_as("nonexistent-dataset", "drb", HTTPStatus.NOT_FOUND)
+        assert response.json == {"message": "Dataset 'nonexistent-dataset' not found"}
+
+    def test_dataset_not_present(self, query_get_as):
+        response = query_get_as("fio_2", "drb", HTTPStatus.NOT_FOUND)
+        assert response.json == {
+            "message": "The dataset tarball named 'random_md5_string4' not found"
+        }
```

Comment:
Having "dataset", "tarball", "name", and "md5" in the same message seems wrong.  From the caller's perspective, the call references a dataset by its ID, right?  The fact that that ID is an MD5 hash which once referred to a tarball is really beside the point.

So, I think this message should be something closer to
```python
        assert response.json == {
            "message": "No dataset with ID 'random_md5_string4' found"
        }
```
or
```python
        assert response.json == {
            "message": "Dataset ID 'random_md5_string4' not found"
        }
```

Reply:
Most likely this is just a server message that's persisted from "the old days" when we really referenced tarballs by name, and it was copied into the new Visualize code. (Which means your comments apply to the original location.)

Still, even without fixing that location right now, improving it here would be good. And, yeah, we identify the dataset by resource ID so that's what we didn't find.